fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1337,6 +1341,170 @@ class GroupSpec {
+                                }
+                                assertEquals(expectedKeys, actualKeys.collect(Collectors.toList()),
+                                        () -> {
+                                            LuceneIndexMaintainer indexMaintainer = (LuceneIndexMaintainer)recordStore.getIndexMaintainer(index);
+                                            return primaryKeys.entrySet().stream().sorted(Map.Entry.comparingByValue())
+                                                    .map(entry -> entry.getValue() + " " + entry.getKey())
+                                                    .collect(Collectors.joining(", ", "All data: [", "]\n")) +
+                                                    indexMaintainer.getPartitioner().getAllPartitionMetaInfo(Tuple.from(1)).join()
+                                                            .stream()
+                                                            .sorted(Comparator.comparing(partitionInfo -> Tuple.fromBytes(partitionInfo.getFrom().toByteArray())))
+                                                            .map(partitionInfo -> partitionInfo.getId() + " (" + partitionInfo.getCount() + "): [" +
+                                                                    Tuple.fromBytes(partitionInfo.getFrom().toByteArray()) + "," +
+                                                                    Tuple.fromBytes(partitionInfo.getTo().toByteArray()) + "]")
+                                                            .collect(Collectors.joining(", ", "Partitions: [", "]"));
+                                        });
+                            }
+                        }
+                    }
+                }
+            }
+            commit(context);
+        }
+    }
+
+    private List<Tuple> queryLocal(Map<Tuple, Long> dataset, Comparisons.Type comparisonType, long comparand, SortType sortType) {
+        List<Map.Entry<Tuple, Long>> hits = dataset.entrySet()
+                .stream()
+                .filter(entry -> {
+                    long value = entry.getValue();
+                    switch (comparisonType) {
+                        case EQUALS:
+                            return value == comparand;
+                        case LESS_THAN:
+                            return value < comparand;
+                        case LESS_THAN_OR_EQUALS:
+                            return value <= comparand;
+                        case GREATER_THAN:
+                            return value > comparand;
+                        case GREATER_THAN_OR_EQUALS:
+                            return value >= comparand;
+                        default:
+                            return false;
+                    }
+                })
+                .collect(Collectors.toList());
+
+        if (sortType == SortType.ASCENDING) {
+            hits.sort((Map.Entry<Tuple, Long> a, Map.Entry<Tuple, Long> b) -> {
```

Comment:
Not a big deal, but I think these sorts could be simplified using something like:
```
Comparator.comparing(entry -> entry.getValue()).thenComparing(Comparator.comparing(entry -> entry.getKey()))
```

And you can use `comparator.reversed()` to do descending.